Core/include/Acts/Propagator/Propagator.hpp

```diff
@@ -59,6 +59,45 @@ struct PropagatorResult : private detail::Extendable<result_list...> {
   double pathLength = 0.;
 };
 
+/// @brief Class holding the trivial options in propagator options
+///
+struct PropagatorPlainOptions {
```

Comment:
This **duplicates** what we have in `PropagatorOptions`. It should **replace** those, no? We also have another duplication in a weird way in the `DensePropagatorOptions` somewhere. Maybe we could use this PR to resolve all of those, ideally, and centralize?

Reply:
Those options in `PropagatorOptions` have not been replaced by this struct is simply because I don't want to change all those related code like `state.options` to something like `state.options.plainOptions`.  Hence, I would prefer to have the `PropagatorOptions` inheriting from this simple struct. This will avoid the duplicated declarations.

However, I'm not very sure what is the problem with the `DenseStepperPropagatorOptions`?

Reply:
Doesn't `DenseStepperPropagatorOptions` not also duplicate these declarations? If no, then nevermind!

Reply:
No, the `DenseStepperPropagatorOptions` is inheriting from the `PropagatorOptions`, hence no duplicated declarations.
I have removed the duplication now.